userd/launcher.go

```diff
@@ -97,7 +99,22 @@ func (s *Launcher) OpenURL(addr string) *dbus.Error {
 		return makeAccessDeniedError(fmt.Errorf("Supplied URL scheme %q is not allowed", u.Scheme))
 	}
 
-	if err = exec.Command("xdg-open", addr).Run(); err != nil {
+	snap, err := snapFromSender(s.conn, sender)
+	if err != nil {
+		return dbus.MakeFailedError(err)
+	}
+
+	xdg_data_dirs := []string{}
+	xdg_data_dirs = append(xdg_data_dirs, fmt.Sprintf("/%s/%s/current/usr/share", snap, dirs.SnapMountDir))
```

Comment:
Wait, isn't that the other way around? 

How about:
```suggestion
	xdg_data_dirs = append(xdg_data_dirs, filepath.Join(dirs.SnapMountDir, snap, "current/usr/share"))
```

Reply:
Doh!  Of course